Application/ResetSystem/ResetSystem.c

```diff
@@ -42,6 +46,47 @@ UefiMain (
     Mode = L"ColdReset";
   }
 
+  if (StrCmp (Mode, L"Firmware") == 0) {
+    DEBUG ((DEBUG_INFO, "OCRST: Entering firmware...\n"));
+    DataSize = sizeof (OsIndications);
+    Status = gRT->GetVariable (
+                    EFI_OS_INDICATIONS_SUPPORT_VARIABLE_NAME,
+                    &gEfiGlobalVariableGuid,
+                    &Attr,
+                    &DataSize,
+                    &OsIndications
+                    );
+    if (!EFI_ERROR (Status)) {
+      if ((OsIndications & EFI_OS_INDICATIONS_BOOT_TO_FW_UI) != 0) {
+        DataSize = sizeof (OsIndications);
+        Status = gRT->GetVariable (
+                        EFI_OS_INDICATIONS_VARIABLE_NAME,
+                        &gEfiGlobalVariableGuid,
+                        &Attr,
+                        &DataSize,
+                        &OsIndications
+                        );
+        if (!EFI_ERROR (Status)) {
+          OsIndications |= EFI_OS_INDICATIONS_BOOT_TO_FW_UI;
+        } else {
+          OsIndications = EFI_OS_INDICATIONS_BOOT_TO_FW_UI;
+        }
+        Status = gRT->SetVariable (
+                        EFI_OS_INDICATIONS_VARIABLE_NAME,
+                        &gEfiGlobalVariableGuid,
+                        EFI_VARIABLE_NON_VOLATILE | EFI_VARIABLE_BOOTSERVICE_ACCESS | EFI_VARIABLE_RUNTIME_ACCESS,
+                        sizeof (OsIndications),
+                        &OsIndications
+                        );
+      }
```

Comment:
Just realised that we want `} else {` with some warning and `Status = EFI_UNSUPPORTED` when `EFI_OS_INDICATIONS_BOOT_TO_FW_UI` is not set. Sorry for not noticing earlier.